Library/Unit/MonthOnWeekDayOfMonthUnit.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace FluentScheduler
+{
+    using System;
+
+    /// <summary>
+    /// Unit of time that represents a specific week day of the month.
+    /// </summary>
+    public sealed class MonthOnWeekDayOfMonthUnit
+    {
+        private readonly int _duration;
+
+        private readonly int weekDayOfMonth;
+
+        internal MonthOnWeekDayOfMonthUnit(Schedule schedule, int duration, int weekDayOfMonth)
```

Comment:
Fluent classes must not see Schedule class. All the work is performed by a calculator.